crates/burn-dataset/src/vision/image_folder.rs

```diff
@@ -468,6 +476,37 @@ impl ImageFolderDataset {
         Self::with_items(items, classes)
     }
 
+    /// Create a COCO detection dataset based on the annotations JSON and image directory.
+    ///
+    /// # Arguments
+    ///
+    /// * `annotations_json` - Path to the JSON file containing annotations in COCO format (for
+    ///   example instances_train2017.json).
+    ///
+    /// * `images_path` - Path containing the images matching the annotations JSON.
+    ///
+    /// # Returns
+    /// A new dataset instance.
+    pub fn new_coco_detection<A: AsRef<Path>, I: AsRef<Path>>(
+        annotations_json: A,
+        images_path: I,
+    ) -> Result<Self, ImageLoaderError> {
+        let file = fs::File::open(annotations_json)
+            .map_err(|e| ImageLoaderError::IOError(format!("Failed to open annotations: {}", e)))?;
+        let json: Value = serde_json::from_reader(file).map_err(|e| {
+            ImageLoaderError::ParsingError(format!("Failed to parse annotations: {}", e))
+        })?;
+
+        let classes = Self::parse_coco_classes(&json)?;
+        let annotations = Self::parse_coco_bbox_annotations(&json)?;
+        let items = Self::parse_coco_images(&images_path, annotations, &json)?;
```

Comment:
Tiny nitpick, but I'd remove the `parse_coco_*` methods from the `ImageFolderDataset` and make them standalone functions (in the same module).

Reply:
Hmm, OK, it's a bit of a mix right now, i.e. `check_extensions()` and `with_items()` are struct members, which was my reasoning for adding the COCO stuff there too, but we can move the COCO helpers out.

Reply:
`with_items(...)` returns an instance of the struct.

Though I agree, `check_extensions` could also be moved out. Feel free to make this change too if you feel like it would make more sense 🙂